examples/cairo_threads/main.rs

```diff
@@ -1,127 +1,41 @@
-            fn as_ref(&self) -> &[u8] {
-                self.image.as_ref().expect("Holding no image").as_ref()
-            }
-        }
-
-        impl AsMut<[u8]> for ImageHolder {
-            fn as_mut(&mut self) -> &mut [u8] {
-                self.image.as_mut().expect("Holding no image").as_mut()
-            }
-        }
-
-        // Temporary move out the pixels
-        let image = self.0.take().expect("Empty image");
+use drawing::{draw_image_if_dirty, draw_initial, draw_slow};
 
-        // A new return location that is then passed to our helper struct below
-        let return_location = Rc::new(RefCell::new(None));
-        {
-            let holder = ImageHolder {
-                image: Some(image),
-                return_location: return_location.clone(),
-            };
-
-            // The surface will own the image for the scope of the block below
-            let surface =
-                ImageSurface::create_for_data(holder, Format::Rgb24, WIDTH, HEIGHT, 4 * WIDTH)
-                    .expect("Can't create surface");
-            func(&surface);
-
-            // Now the surface will be destroyed and the pixels are stored in the return_location
-        }
+fn main() {
+    let application = gtk::Application::new(
+        Some("com.github.gtk-rs.examples.cairo_threads"),
+        Default::default(),
+    )
+    .expect("Initialization failed...");
 
-        // And here move the pixels back again
-        self.0 = Some(
-            return_location
-                .borrow_mut()
-                .take()
-                .expect("Image not returned"),
-        );
-    }
+    application.connect_activate(build_ui);
+    application.run(&args().collect::<Vec<_>>());
 }
 
-// This example runs four worker threads rendering parts of the image independently at different
-// paces in a sort of double buffered way.
-//
-// +---+---+
-// | 0 | 1 |
-// +---+---+
-// | 2 | 3 |
-// +---+---+
-//
-// Each worker thread waits for an image to render into, sleeps for a while, does the drawing, then
-// sends the image back and waits for the next one.
-//
-// The GUI thread holds an image per image part at all times and these images are painted on a
-// DrawingArea in its 'draw' signal handler whenever needed.
-//
-// Additionally the GUI thread has a channel for receiving the images from the worker threads. If
-// there is a new image, the old image stored by the GUI thread is replaced with the new one and
-// the old image is sent back to the worker thread. Then the appropriate part of the DrawingArea is
-// invalidated prompting a redraw.
-//
-// The two images per thread are allocated and initialized once and sent back and forth repeatedly.
-
 fn build_ui(application: &gtk::Application) {
     let window = ApplicationWindow::new(application);
     let area = DrawingArea::new();
     window.add(&area);
 
-    area.set_size_request(WIDTH * 2, HEIGHT * 2);
+    let (width, height) = (200, 200);
```

Comment:
Why replace these constants with hardcoded values here? Having them being passed to the functions as arguments makes sense, but in this function at least making direct use of constants seems cleaner.

Reply:
Sure, I can readd the constant